### ISSUE TITLE
UnmarshalText issue

### ISSUE THREAD

**Issue by bilus** [gorilla/schema#100]
Great library, thank you!

I'm having a problem making UnmarshalText be nice and cooperate. Here's simplest possible code recreating the issue:

```golang
package main

import (
	"fmt"
	"github.com/gorilla/schema"
)

type Email struct {
	Address string
}

func (e *Email) UnmarshalText(text []byte) (err error) {
	s := string(text)
	e.Address, err = s, nil
	return
}

type Person struct {
	Name   string
	Emails []Email
}

func main() {
	v := map[string][]string{
		"name":   {"John"},
		"emails": {"foo", "bar"},
	}
	var p Person

	err := schema.NewDecoder().Decode(&p, v)
	if err != nil {
		fmt.Println(err)
	}

	fmt.Println(p)
}
```

Output:

```
schema: invalid path "emails"
{John []}
```

Is there anything broken or (more likely!) I'm doing sth wrong?

**Comment by bilus** [comment]
UPDATE: The following works (using single email instead of a slice):

```golang
package main

import (
	"fmt"
	"github.com/gorilla/schema"
)

type Email struct {
	Address string
}

func (e *Email) UnmarshalText(text []byte) (err error) {
	s := string(text)
	e.Address, err = s, nil
	return
}

type Person struct {
	Name   string
	Emails []Email
	Email <- change
}

func main() {
	v := map[string][]string{
		"name": {"John"},
		// "emails": {"foo", "bar"}, // <- changes
		"email": {"foo"},  // <- changes
	}
	var p Person

	err := schema.NewDecoder().Decode(&p, v)
	if err != nil {
		fmt.Println(err)
	}

	fmt.Println(p)
}
```

**Comment by bilus** [comment]
I THINK I kinda understand what is going on here. Can someone verify my way of thinking?

TL;DR Because the lib thinks this is a struct, it assumes I'll be using syntax for using structs, e.g. (`emails.0=xxx`). My use case is this:

Given the following query string `?items=foo.bar,baz.qux` I want to split each item on `.` and create a struct, e.g. 

```golang
type Item struct {
  field0 string
  field1 string
}
```

Does the use case make sense at all? Should I use `UnmarshalText` for this or sth else?
